connect/tls.go

```diff
@@ -366,7 +366,22 @@ func (cfg *dynamicTLSConfig) Leaf() *tls.Certificate {
 func (cfg *dynamicTLSConfig) Ready() bool {
 	cfg.RLock()
 	defer cfg.RUnlock()
-	return cfg.leaf != nil && cfg.roots != nil
+	if cfg.leaf == nil || cfg.roots == nil {
+		return false
+	}
+
+	if cfg.leaf.Leaf == nil {
+		if cert, err := x509.ParseCertificate(cfg.leaf.Certificate[0]); err == nil {
+			cfg.leaf.Leaf = cert
```

Comment:
This is modifying state while only holding an `RLock`. Is that dangerous?

Reply:
Should we update the comment on ReadyWait to be strong than "which we assume are valid"? Happy to leave it as only triggering on the first ready state change rather than being a more elaborate watch mechanism.